src/register.jl

```diff
@@ -0,0 +1,166 @@
+
+    return im1_padded, im2_padded
+end
+
+"""
+Computes the shape difference between im_reference and im_target for each angle in test_angles.
+The reference image is held constant, while the target image is rotated. The test_angles are interpreted
+as the angle of rotation from target to reference, so to find the best match, we rotate the reverse
+direction. A perfect match at angle `A` would imply im_target is the same shape as if im_reference was
+rotated by `A`. 
+Use `imrotate_function=imrotate_bin_<clockwise|counterclockwise>_<radians|degrees>` to get angles <clockwise|counterclockwise> in <radians|degrees>.
+"""
+function shape_difference_rotation(im_reference, im_target, test_angles; imrotate_function=imrotate_bin_clockwise_radians)
+    shape_differences = Array{
+        NamedTuple{(:angle, :shape_difference),Tuple{Float64,Float64}}
+    }(
+        undef, length(test_angles)
+    )
+
+    for (idx, angle) in enumerate(test_angles)
+
+        # rotate image back by angle
+        im_target_rotated = imrotate_function(im_target, -angle)
+
+        im1, im2 = align_centroids(im_reference, im_target_rotated)
+
+        # Check here that im1 and im2 sizes are the same
+        # This should be guaranteed by "align_centroids"
+        @assert size(im1) == size(im2)
+
+        a_not_b = im1 .> 0 .&& isequal.(im2, 0)
+        b_not_a = im2 .> 0 .&& isequal.(im1, 0)
+        shape_difference = sum(a_not_b .|| b_not_a)
+        shape_differences[idx] = (; angle, shape_difference)
+
+    end
+    return shape_differences
+end
+
+
+"""
+The default registration angles are evenly distributed in steps of π/36 rad (5º) around a full rotation,
+ensuring that no angles are repeated (since -π rad == π rad),
+and ordered so that smaller absolute angles which are positive will be returned in the event of a tie in the shape difference.
+"""
+register_default_angles_rad = sort(reverse(range(; start=-π, stop=π, step=π / 36)[1:(end-1)]); by=abs)
+
+"""
+Finds the image rotation angle in `test_angles` which minimizes the shape difference between `im_reference` and `im_target`.
+The default test angles are shown in `register_default_angles_rad`.
+Use `imrotate_function=imrotate_bin_<clockwise|counterclockwise>_<radians|degrees>` to get angles <clockwise|counterclockwise> in <radians|degrees>.
+"""
+function register(
+    im_reference,
+    im_target;
+    test_angles=register_default_angles_rad,
+    imrotate_function=imrotate_bin_clockwise_radians,
+)
+    shape_differences = shape_difference_rotation(im_reference, im_target, test_angles; imrotate_function)
+    best_match = argmin((x) -> x.shape_difference, shape_differences)
+    return best_match.angle
+end
+
+"""
+The default registration angles are evenly distributed in steps of 5º around a full rotation,
+ensuring that no angles are repeated (since -180º == -180º),
+and ordered so that smaller absolute angles which are positive will be returned in the event of a tie in the shape difference.
```

Comment:
Oh okay - so argmin defaults to just returning the first instance?

Reply:
Yes:
```julia
julia> argmin((x)->x.v, [(;id=1, v=1), (;id=2, v=1), (;id=3, v=1)])
(id = 1, v = 1)
```